### PR TITLE
chore: move `build,build_extended` logic to lib

### PR DESCRIPTION
**Pull request by dj8yfo**
`builder` field in ABI json now looks like this:

```json
...
{
  "schema_version": "0.4.0",
  "metadata": {
    "name": "donation-sample-single-package-no-workspace",
    "version": "0.7.7",
    "build": {
      "compiler": "rustc 1.80.1",
      "builder": "cargo-near cargo-near-build 0.1.0"
    }
  },
...  
```
---

interop of older 0.6.3 build script with docker image from pr - [Cargo.toml](https://github.com/dj8yfo/factory-rust/blob/6ed50ca865d83f7281b99afda6ea2181a9a77176/factory/Cargo.toml#L13-L25)
(`cargo check` runs in `  Finished dev profile [unoptimized + debuginfo] target(s) in 1m 03s`

older_build_rs_in_newer_image

```bash
sha256sum *.wasm
2d4a6cd07d59aeacefd7ed73aa58c106bbf23f0b10dd8f56960944fe785cee90  donation-product.repro-fct-45.testnet.wasm
f2e956ed163039c0f341a5048d9260217d4ec0d685fcf1e497222e4234bb47a5  repro-fct-45.testnet.wasm
2d4a6cd07d59aeacefd7ed73aa58c106bbf23f0b10dd8f56960944fe785cee90  repro-fct-product-45.testnet.wasm
```

---
interop of newer 0.1.0 cargo-near-build build script (from pr) with older docker image - [Cargo.toml](https://github.com/dj8yfo/factory-rust/blob/29b5e34014a7a3494c009739f2005afcfc7ac282/factory/Cargo.toml#L13-L25)

( `cargo check` runs in `    Finished dev profile [unoptimized + debuginfo] target(s) in 11.62s`)

older_image_with_new_build_rs

```bash
sha256sum *.wasm
ad162718dd5cd43d46e5fb8db90f59bf777940def53884c4287a8f8b6f16b054  donation-product.repro-fct-46.testnet.wasm
cf5b9dc7c9c01a28fcb2209339f01d91f3edc144a20714626c1087aa9ee23e00  repro-fct-46.testnet.wasm
ad162718dd5cd43d46e5fb8db90f59bf777940def53884c4287a8f8b6f16b054  repro-fct-product-46.testnet.wasm

```

### CODE REVIEW

**Comment by dj8yfo** [comment]
@race-of-sloths ping

**Comment by race-of-sloths** [comment]
@dj8yfo Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Weekly streak is on the road, smart strategy! Secure your streak with another PR!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/dj8yfo)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/dj8yfo)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Review by frol — APPROVED** [review]
@dj8yfo great job! Once we will approach the final stage, we'd need to review all the public structs and functions and minimize their number to keep the public interface minimal

@race-of-sloths score 13

**Pull request merged**
